src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -140,17 +140,17 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
                                 noOpTelemetry.StartIntervalMeasure();
 
                                 // Replay Warnings and Errors from an existing lock file in case of a no-op.
-                                await MSBuildRestoreUtility.ReplayWarningsAndErrorsAsync(_request.ExistingLockFile, _logger);
+                                await MSBuildRestoreUtility.ReplayWarningsAndErrorsAsync(cacheFile.LogMessages, _logger);
 
                                 noOpTelemetry.EndIntervalMeasure(ReplayLogsDuration);
 
                                 restoreTime.Stop();
 
                                 return new NoOpRestoreResult(
                                     _success,
-                                    _request.ExistingLockFile,
-                                    _request.ExistingLockFile,
-                                    _request.ExistingLockFile.Path,
+                                    null,
```

Comment:
Why null? If they are not needed maybe the NoOpRestoreResult needs to be modified to remove them from constructor?

Reply:
I am not familiar enough with this object to know, does someone on the NuGet team want to tell me if I can remove these inputs from the constructor?

Reply:
I think we should not pass `null`.  Lazy seems like a safe bet.  There are places (like src\NuGet.Core\NuGet.PackageManagement\NuGetPackageManager.cs) that check `RestoreResult.LockFile` and `RestoreResult.PreviousLockFile` and having them now return `null` seems like an undesirable change.

Reply:
Okay this is done